lib/Crypto/Crypt.php

```diff
@@ -226,10 +239,11 @@ public function generateHeader($keyFormat = 'hash') {
 	 * @throws EncryptionFailedException
 	 */
 	private function encrypt($plainContent, $iv, $passPhrase = '', $cipher = self::DEFAULT_CIPHER) {
+		$options = $this->useLegacyEncoding ? 0 : OPENSSL_RAW_DATA;
```

Comment:
We need to have consistent checks for the variable or ensure the variable is a boolean. If the admin enters "true" as string (or any other random string), the "encoding" header will be set as binary, but the contents will be written as base64.

I don't care too much about the final decision in the above scenario (we're expecting a boolean anyway), but it mustn't break the file.

Reply:
I have added strict comparison here. Also, `useLegacyEncoding` variable now should be consistently boolean.